[BUG] [cpp-qt] Unable to run cmake and use Qt5 anymore

#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. Eg:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description

I am unable to run the generator after updating to a newer version (7.2.0) and using Qt5. I didn't change anything about my Qt installation, just the openapi-generator version, and ran into failures. (I also upgraded CMake just to be sure).

I set CMAKE_PREFIX_PATH before running like this (to set my Qt version):

```
export CMAKE_PREFIX_PATH=C:\\Qt\\5.15.16\\msvc2019_64\\lib\\cmake
cmake  ../src/client/
```

I get this:

```
-- Building for: Visual Studio 17 2022
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Selecting Windows SDK version 10.0.20348.0 to target Windows 10.0.22631.
-- The C compiler identification is MSVC 19.38.33134.0
-- The CXX compiler identification is MSVC 19.38.33134.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning (dev) at CMakeLists.txt:5 (find_package):
  Policy CMP0084 is not set: The FindQt module does not exist for
  find_package().  Run "cmake --help-policy CMP0084" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at C:/Program Files/CMake/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Qt3 (missing: QT_QT_LIBRARY QT_INCLUDE_DIR
  QT_MOC_EXECUTABLE)
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/CMake/share/cmake-3.28/Modules/FindQt3.cmake:213 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  C:/Program Files/CMake/share/cmake-3.28/Modules/FindQt.cmake:160 (include)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!

```

If I add `find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core Network Gui)` back in, all works fine and generates

@MartinDelille do you remember why in https://github.com/OpenAPITools/openapi-generator/pull/14968 this line was removed?

--- a/modules/openapi-generator/src/main/resources/cpp-qt-client/CMakeLists.txt.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-qt-client/CMakeLists.txt.mustache
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.5)
 
 project({{{packageName}}})
 
+set(CMAKE_AUTOMOC ON)
+
 find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core Network Gui)
 find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core Network Gui)
 

--- a/modules/openapi-generator/src/main/resources/cpp-qt-client/CMakeLists.txt.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-qt-client/CMakeLists.txt.mustache
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 3.5)
 
 project({{{packageName}}})
 
+find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core Network Gui)
 find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core Network Gui)
 
 include(GNUInstallDirs)

--- a/samples/client/petstore/cpp-qt/client/CMakeLists.txt
+++ b/samples/client/petstore/cpp-qt/client/CMakeLists.txt
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.5)
 
 project(CppQtPetstoreClient)
 
+set(CMAKE_AUTOMOC ON)
+
 find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core Network Gui)
 find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core Network Gui)
 

--- a/samples/client/petstore/cpp-qt/client/CMakeLists.txt
+++ b/samples/client/petstore/cpp-qt/client/CMakeLists.txt
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 3.5)
 
 project(CppQtPetstoreClient)
 
+find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core Network Gui)
 find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core Network Gui)
 
 include(GNUInstallDirs)

